### PR TITLE
Fix ros2 parameter descriptions and range values

### PR DESCRIPTION
**Pull request by SamerKhshiboun**
Tracked by [DSO-18759]
- Added the correct step value to the Integer Range step value.
- Added default value to the parameter description section.
- Added parameter type
- Did some manipulation on floats before converting them to double, in order to get more precise values Thanks to @maloel suggestion
- Now, all warnings due to parameters wrong ranges or values are solved.
- Split long functions into short ones

### CODE REVIEW

**Comment by SamerKhshiboun** [comment]
Don't merge yet.
Might change some other parts related to other warnings (not related to this ticket) but in the same area.

**Comment by christian-rauch** [comment]
> Don't merge yet.

You can convert a PR into a draft (see `Convert to draft` next to `Still in progress?` in the top right) to disable merging. Once you are ok with the PR, you can remove the "draft" state.

File: realsense2_camera/src/sensor_params.cpp
Comment (Nir-Az) [inline realsense2_camera/src/sensor_params.cpp]: != 0?

File: realsense2_camera/src/sensor_params.cpp
Comment (Nir-Az) [inline realsense2_camera/src/sensor_params.cpp]: Maybe needs conversion too?

File: realsense2_camera/src/sensor_params.cpp
Comment (Nir-Az) [inline realsense2_camera/src/sensor_params.cpp:97]: Please protect div_by_zero. for both devision operations

**Review by Nir-Az — APPROVED** [review]
Great!

File: realsense2_camera/src/sensor_params.cpp
Comment (Nir-Az) [reply realsense2_camera/src/sensor_params.cpp:97]: @SamerKhshiboun we forgot about it before merging  please add protection
I wonder why cppcheck didn't shout on it..

**Pull request merged**
